### PR TITLE
Inject inferred queue settings into agent config generation

### LINKED ISSUE #26638: Support customization of throughput relevant settings in APM integration
APM Server standalone allows users to customize throughput relevant settings:
* `output.elasticsearch.worker`, `output.elasticsearch.bulk_max_size`
* `queue.mem.events`, `queue.mem.flush.min_events`

Elastic Agent with Fleet does not expose the memory queue related settings, but they are important to be aligned with the available resources to optimize processing while not running out of memory.
A discussed approach to support this is to make following changes in libbeat:
* ensure that beats properly restart/reload when `output` related settings change (`output.elasticsearch.worker`, `output.elasticsearch.bulk_max_size`)
* introduce a factor `F` that can be set per beat (not user editable); 
  * `F=1`: lowest throughput, no load balancing.
  * `F = (W + R)` with `W = len(output.*.hosts) * worker`. `R` sets how many batches can be ready for outputs still waiting for ACKs. `R=W`has higher memory usage, but can keep the outputs/queues more saturated improving throughput (otherwise it has to wait for the next N events to be acquired). For reducing memory usage set `R=0` or `R=1`. If dealing with bursts of events, `F=1` might make sense. Otherwise `F=2*W` could be a good default. 
 
* translate libbeat internal `queue` settings from output relevant settings and `F`: 
  * `queue.mem.flush.min_events` = `bulk_max_size`
  *  `queue.mem.events` = `bulk_max_size * F`
  * set a fixed value for `queue.mem.flush.timeout`

cc @urso

### PR DESCRIPTION
**Pull request by faec**
## What does this PR do?

Add a new internal rule, `inject_queue`, to agent's config transpilation process, which derives implicit memory queue parameters from the user-specified output settings and injects them into the generated configuration, and call the rule from the appropriate spec files. See #26638 for details on the reasoning behind this feature.

I haven't changed the documentation or default config files, since this change doesn't add any new user-customizable settings, it just adapts the backend settings better to the context. There should be a followup PR that clarifies that queue behavior is affected by output settings `bulk_max_size` and `worker`, although we're intentionally not committing to an exact conversion formula since we may want to refine it in the future.

Resolves #26638.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/elastic-agent/pkg/agent/transpiler/rules.go [hunk 1]
```diff
@@ -1519,6 +1523,107 @@ func InsertDefaults(path string, selectors ...Selector) *InsertDefaultsRule {
...
+		return nil
+	}
+
+	outputDict, ok := outputNode.Value().(*Dict)
+	if !ok || len(outputDict.value) == 0 {
+		faeNode := &Key{
+			name:  "hello",
+			value: &StrVal{value: "fae was here"},
+		}
+		if err := Insert(ast, faeNode, "output"); err != nil {
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/transpiler/rules.go:1551]: Should this be here at all? Was this a debugging thing?

I think if no output is defined it should just return nil and do nothing.

File: x-pack/elastic-agent/pkg/agent/transpiler/rules.go [hunk 2]
```diff
@@ -1519,6 +1523,107 @@ func InsertDefaults(path string, selectors ...Selector) *InsertDefaultsRule {
...
+			},
+			&Key{
+				name: "flush",
+				value: &Dict{
+					value: []Node{
+						&Key{
+							name:  "min_events",
+							value: &IntVal{value: min_events},
+						},
+						// flush.timeout defaults to 1s, which is what we want for now,
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/transpiler/rules.go:1600]: should we always set the default by Elastic Agent? that way if libbeat changes that doesn't affect what Elastic Agent expects?

no hard requirement to change this behavior, its just a thought

**Review by blakerouse — CHANGES_REQUESTED** [review]
I think one part needs to be fixed. Otherwise looks good.

File: x-pack/elastic-agent/pkg/agent/transpiler/rules.go (hunk 1 above)
Comment (faec) [reply x-pack/elastic-agent/pkg/agent/transpiler/rules.go]: Haha yes, leftover from when I was still figuring out how to traverse ASTs, removed

File: x-pack/elastic-agent/pkg/agent/transpiler/rules.go (hunk 2 above)
Comment (faec) [reply x-pack/elastic-agent/pkg/agent/transpiler/rules.go]: I debated that... this has been a stable parameter but yea, let's make it explicit. Done.

**Review by blakerouse — APPROVED** [review]
Thanks for the fixes. Looks good.

**Pull request merged**